shared/fs/index.js

```diff
@@ -64,29 +63,35 @@ const FileRow = RowConnector(({path, name, icon, onOpen}: FileRowProps) => (
   <ClickableBox onClick={onOpen} style={{...stylesCommonRow}}>
     <Box style={{...globalStyles.flexBoxRow, alignItems: 'center', flex: 1}}>
       <Icon type={icon} style={{marginRight: globalMargins.small}} />
-      <Box style={FolderBoxStyle}>
+      <Box style={folderBoxStyle}>
         <Text type="Body">{name}</Text>
       </Box>
     </Box>
   </ClickableBox>
 ))
 
-const Files = ({path, items, onBack}: FolderProps) => (
-  <Box style={styleOuterContainer}>
-    <Box style={stylesContainer}>
-      {onBack &&
-        Types.pathToString(path) !== '/keybase' && (
-          <Box style={globalStyles.flexBoxColumn}>
-            <BackButton onClick={onBack} style={{left: 16, position: 'absolute', top: 16}} />
-          </Box>
-        )}
-      <FolderHeader title={'Folders: ' + Types.pathToString(path)} />
-      <List
-        items={items}
-        renderItem={(index, item) => <FileRow key={Types.pathToString(item)} path={item} />}
-      />
-    </Box>
-  </Box>
-)
+class Files extends React.PureComponent<FolderProps> {
+  _renderRow = (index, item) => <FileRow key={Types.pathToString(item)} path={item} />
+
+  render() {
+    const onBack = this.props.onBack
```

Comment:
why pull these out?
if they're re-used a bunch of times in here maybe but if there used just once just refer to it as this.props.onBack etc
or
do
```
const {onBack, path, items} = this.props
```
but even then i'd just extract things that are used a bunch